### PR TITLE
mgr/cephadm: Make `ceph orch * --refresh` asynchronous

### PR DESCRIPTION
**Pull request by sebastian-philipp**
There is just no way we can allow those calls to be synchornous:

* All CLI commands are blocked, till the refresh completes.
* ^C doesn't help at all, as it just kills the client

See https://docs.ceph.com/docs/master/dev/cephadm/#note-regarding-network-calls-from-cli-handlers for details

Therefore they have to be scheduled in the bg.

Fixes: https://tracker.ceph.com/issues/46813

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

This change affects ceph-linode and mgr/dashboard as well

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -1668,17 +1680,19 @@ def get_inventory(self, host_filter=None, refresh=False) -> List[orchestrator.In
           - add filtering by label
         """
         if refresh:
-            # ugly sync path, FIXME someday perhaps?
```
Comment (jschmid1) [inline src/pybind/mgr/cephadm/module.py:1671]: seems like `someday` is today :tada:

**Comment by sebastian-philipp** [comment]
green: https://pulpito.ceph.com/swagner-2020-08-05_08:10:39-rados:cephadm-wip-swagner-testing-2020-08-04-1550-distro-basic-smithi/

**Comment by jschmid1** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2020-08-06_14:10:14-rados:cephadm-wip-swagner-testing-2020-08-06-1324-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Pull request merged**